command/agent/dns.go

```diff
@@ -525,9 +538,17 @@ func syncExtra(index map[string]dns.RR, resp *dns.Msg) {
 // 1035. Enforce an arbitrary limit that can be further ratcheted down by
 // config, and then make sure the response doesn't exceed 512 bytes. Any extra
 // records will be trimmed along with answers.
-func trimUDPResponse(config *DNSConfig, resp *dns.Msg) (trimmed bool) {
+func trimUDPResponse(config *DNSConfig, req, resp *dns.Msg) (trimmed bool) {
```

Comment:
Actually, if you pass `maxSize` in here instead of `req` this would  be simpler to test on its own.